### PR TITLE
Scan build fixes

### PR DESCRIPTION
**Pull request by hjmjohnson**
Ran scan-build on code subset that is included with ITK.

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@dzenanz  some VXL cleanups for new clang compilers and scan-build identified issues.

File: core/vnl/algo/vnl_lsqr.h [hunk 1]
```diff
@@ -70,7 +70,7 @@ class VNL_ALGO_EXPORT vnl_lsqr
   double resid_norm_estimate_;
   double result_norm_estimate_;
   double A_condition_estimate_;
-  double result_norm_;
+  /* Never used double result_norm_; */
```
Comment (dzenanz) [inline core/vnl/algo/vnl_lsqr.h:73]: Remove commented out code?

File: .gitignore [hunk 2]
```diff
@@ -49,3 +49,4 @@ cmake-build-*/
 
 # Exclude to keep j2k implementation separate from vxl
 v3p/j2k/*
+/CMakeUserPresets.json
```
Comment (dzenanz) [inline .gitignore:52]: Does it need a relative path?

File: vcl/tests/test_memory.cxx [hunk 3]
```diff
@@ -60,13 +60,13 @@ int test_memory_main(int /*argc*/,char* /*argv*/[])
     ASSERT(!pa0.get(), "auto_ptr holds an object after release()");
 
     pa1 = std::move(pa3);
-    ASSERT(!pa3.get(), "auto_ptr holds an object after assignment to another");
+    ASSERT(!pa1.get(), "auto_ptr holds an object after assignment to another");
```
Comment (dzenanz) [inline vcl/tests/test_memory.cxx]: Shouldn't exactly one of these fail? Line 63 vs 64.

File: core/vnl/vnl_bignum.cxx [hunk 4]
```diff
@@ -19,7 +19,7 @@ using Data = unsigned short;
 
 vnl_bignum::vnl_bignum()
 
-    = default;
+  = default;
```
Comment (dzenanz) [inline core/vnl/vnl_bignum.cxx:22]: Shouldn't this be moved to line 20?

File: vcl/tests/test_memory.cxx [hunk 5]
```diff
@@ -60,7 +60,6 @@ int test_memory_main(int /*argc*/,char* /*argv*/[])
     ASSERT(!pa0.get(), "auto_ptr holds an object after release()");
 
     pa1 = std::move(pa3);
-    ASSERT(!pa1.get(), "auto_ptr holds an object after assignment to another");
```
Comment (dzenanz) [inline vcl/tests/test_memory.cxx]: This removal should be squashed into the earlier commit.

**Review by dzenanz — APPROVED** [review]

**Pull request merged**